src/components/Facility/AssetCreate.tsx

```diff
@@ -182,180 +194,166 @@ const AssetCreate = (props: AssetProps) => {
-            invalidForm = true;
-          }
-          return;
-        case "is_working":
-          if (is_working === undefined) {
-            errors[field] = t("field_required");
-            invalidForm = true;
-          }
-          return;
-        case "location":
-          if (!location || location === "0" || location === "") {
-            errors[field] = "Select a location";
-            invalidForm = true;
-          }
-          return;
-        case "support_phone": {
-          if (!support_phone) {
-            errors[field] = t("field_required");
-            invalidForm = true;
-          }
-          // eslint-disable-next-line no-case-declarations
-          const checkTollFree = support_phone.startsWith("1800");
-          const supportPhoneSimple = support_phone
-            .replace(/[^0-9]/g, "")
-            .slice(2);
-          if (supportPhoneSimple.length != 10 && !checkTollFree) {
-            errors[field] = "Please enter valid phone number";
-            invalidForm = true;
-          } else if (
-            (support_phone.length < 10 || support_phone.length > 11) &&
-            checkTollFree
-          ) {
-            errors[field] = "Please enter valid phone number";
-            invalidForm = true;
-          }
-          return;
-        }
-        case "support_email":
-          if (support_email && !validateEmailAddress(support_email)) {
-            errors[field] = "Please enter valid email id";
-            invalidForm = true;
-          }
-          return;
-        case "last_serviced_on":
-          if (notes && !last_serviced_on) {
-            errors[field] = "Last serviced on date is require with notes";
-            invalidForm = true;
-          }
-          return;
-        default:
-          return;
-      }
-    });
-    if (invalidForm) {
-      dispatch({ type: "set_error", errors });
-      const firstError = Object.keys(errors).find((key) => errors[key]);
-      if (firstError) {
-        fieldRef[firstError].current?.scrollIntoView({
-          behavior: "smooth",
-          block: "center",
-        });
+  const AssetFormValidator = (form: AssetData): FormErrors<AssetData> => {
+    const errors: Partial<Record<keyof AssetData, FieldError>> = {}; // Initialize error object
+
+    errors.name = RequiredFieldValidator()(form.name);
+
+    if (form.is_working === undefined) {
+      errors.is_working = t("field_required");
+    }
+
+    if (!form.location || form.location === "0" || form.location === "") {
+      errors.location = t("select_local_body");
+    }
+
+    if (!form.support_phone) {
+      errors.support_phone = t("field_required");
+    } else {
+      const checkTollFree = form.support_phone.startsWith("1800");
+      const supportPhoneSimple = form.support_phone
+        .replace(/[^0-9]/g, "")
+        .slice(2);
+      if (supportPhoneSimple.length !== 10 && !checkTollFree) {
+        errors.support_phone = "Please enter valid phone number";
```

Comment:
Add translations (here and elsewhere in this file). @rajku-dev 

Check en.json if there's already a value, if not add 👍